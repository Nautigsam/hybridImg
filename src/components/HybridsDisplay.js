import React from 'react';
import { connect } from 'react-redux';
import { Tile } from 'react-bulma-components';

import HybridDisplay from './HybridDisplay';

const HybridsDisplay = ({ hybrids }) => {
  const hybridByLine = 4; // Must be 1, 2, 3, 4, 6 or 12

  return (
    <>
      <br />
      <Tile className="is-ancestor">
        <Tile style={{ flexWrap: 'wrap' }}>
          {hybrids.map(hybrid => (
            <Tile
              key={hybrids.indexOf(hybrid)}
              className={`is-parent is-${12 / hybridByLine}`}
            >
              <Tile className="is-child">
                <HybridDisplay hybridId={hybrid.id} />
              </Tile>
            </Tile>
          ))}
        </Tile>
      </Tile>
    </>
  );
};

function mapStateToProps(state) {
  return {
    hybrids: state.hybrids,
  };
}

const ConnectedHybridsDisplay = connect(mapStateToProps)(HybridsDisplay);
export default ConnectedHybridsDisplay;
