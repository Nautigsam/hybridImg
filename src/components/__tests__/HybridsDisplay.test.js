import React from 'react';
import { shallow } from 'enzyme';
import { HybridsDisplay } from '../HybridsDisplay';

describe('HybridsDisplay', () => {
  it('renders correctly', () => {
    const props = {
      hybrids: [{ id: 'H1' }, { id: 'H2' }],
    };
    const wrapper = shallow(<HybridsDisplay {...props} />);
    expect(wrapper.children().length).toBe(2);
    expect(wrapper).toMatchSnapshot();
  });
});
