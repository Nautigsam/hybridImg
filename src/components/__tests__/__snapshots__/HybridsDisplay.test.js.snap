// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HybridsDisplay renders correctly 1`] = `
<Fragment>
  <br />
  <Tile
    className="is-ancestor"
    notification={false}
    renderAs="div"
    vertical={false}
  >
    <Tile
      notification={false}
      renderAs="div"
      vertical={false}
    >
      <Tile
        className="is-vertical"
        key="0"
        notification={false}
        renderAs="div"
        vertical={false}
      >
        <Tile
          className="is-parent"
          key="H1"
          notification={false}
          renderAs="div"
          vertical={false}
        >
          <Tile
            className="is-child"
            notification={false}
            renderAs="div"
            vertical={false}
          >
            <HybridDisplay
              hybrid={
                Object {
                  "id": "H1",
                }
              }
            />
          </Tile>
        </Tile>
      </Tile>
      <Tile
        className="is-vertical"
        key="1"
        notification={false}
        renderAs="div"
        vertical={false}
      >
        <Tile
          className="is-parent"
          key="H2"
          notification={false}
          renderAs="div"
          vertical={false}
        >
          <Tile
            className="is-child"
            notification={false}
            renderAs="div"
            vertical={false}
          >
            <HybridDisplay
              hybrid={
                Object {
                  "id": "H2",
                }
              }
            />
          </Tile>
        </Tile>
      </Tile>
      <Tile
        className="is-vertical"
        key="2"
        notification={false}
        renderAs="div"
        vertical={false}
      />
      <Tile
        className="is-vertical"
        key="3"
        notification={false}
        renderAs="div"
        vertical={false}
      />
    </Tile>
  </Tile>
</Fragment>
`;
